src/vector_of_array.jl

```diff
@@ -905,7 +905,9 @@ for (type, N_expr) in [
                 else
                     unpacked = unpack_voa(bc, i)
                     arr_type = StaticArraysCore.similar_type(dest[:, i])
-                    dest[:, i] = if length(unpacked) == 1
+                    dest[:, i] = if length(unpacked) == 1 && length(dest[:, i]) == 1
+                        arr_type(unpacked[1])
```

Comment:
I added one more condition to deal with our special case - it worked in my place please rerun CI @ChrisRackauckas 

There might be room for improvement, but for now, I just need to make it pass all tests.